src/command_modules/azure-cli-botservice/azure/cli/command_modules/botservice/azure/azure_region_mapper.py

```diff
@@ -0,0 +1,54 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from knack.log import get_logger
+
+logger = get_logger(__name__)
+
+
+class AzureRegionMapper:
+
+    @staticmethod
+    def get_app_insights_location(key):
+        region_map = {
+            'australiaeast': 'southeastasia',
+            'australiacentral': 'southeastasia',
+            'australiacentral2': 'southeastasia',
+            'australiasoutheast': 'southeastasia',
+            'eastasia': 'southeastasia',
+            'southeastasia': 'westus',
+            'eastus': 'eastus',
+            'eastus2': 'eastus',
+            'southcentralus': 'southcentralus',
+            'westcentralus': 'westus2',
+            'westus': 'westus2',
+            'westus2': 'westus2',
+            'brazilsouth': 'southcentralus',
+            'centralus': 'southcentralus',
+            'northcentralus': 'southcentralus',
+            'japanwest': 'southeastasia',
+            'japaneast': 'southeastasia',
+            'southindia': 'southeastasia',
+            'centralindia': 'southeastasia',
+            'westindia': 'southeastasia',
+            'canadacentral': 'southcentralus',
+            'canadaeast': 'eastus',
+            'koreacentral': 'southeastasia',
+            'koreasouth': 'southeastasia',
+            'northeurope': 'northeurope',
+            'westeurope': 'westeurope',
+            'uksouth': 'westeurope',
+            'ukwest': 'westeurope',
+            'francecentral': 'westeurope',
+            'francesouth': 'westeurope'
+        }
+        region = region_map.get(key)
```

Comment:
What is the purpose of this?

Reply:
(Relevant issue: Microsoft/botbuilder-tools#543)

This code I believe is to map possible bot-deployment regions to the closest Application Insights region to improve latency.